samcli/lib/providers/sam_stack_provider.py

```diff
@@ -212,15 +226,16 @@ def get_stacks(
         global_parameter_overrides: Optional[Dict]
             Optional dictionary of values for SAM template global parameters
             that might want to get substituted within the template and its child templates
-
+        root_template_dir: str
+            Optional directory of root SAM Template. By default it is None unless we pass in a value.
         Returns
         -------
         stacks: List[Stack]
             The list of stacks extracted from template_file
         remote_stack_full_paths : List[str]
             The list of full paths of detected remote stacks
         """
-        template_dict = get_template_data(template_file)
+        template_dict = get_template_data(template_file, root_template_dir)
```

Comment:
Do we want to provide `root_template_dir` for all usage of `get_template_data()`? Maybe we can make it not optional

Reply:
Currently it is used in a few other places including intrinsics resolver. I prefer not changing that part. Ideally as Chris said, we can get rid of all `root_template_dir` and convert all paths to absolute in a single place later.